File: Backlogr.Api/Models/Entities/Game.cs
namespace Backlogr.Api.Models.Entities;

public sealed class Game
{
    public Guid GameId { get; set; }

    public long? IgdbId { get; set; }

    public string Title { get; set; } = string.Empty;

    public string? Slug { get; set; }

    public string? Summary { get; set; }

    public string? CoverImageUrl { get; set; }

    public DateTime? ReleaseDate { get; set; }

    public string? Developer { get; set; }

    public string? Publisher { get; set; }

    public string? Platforms { get; set; }

    public string? Genres { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<GameLog> GameLogs { get; set; } = new List<GameLog>();
}

File: Backlogr.Web/types/game.ts
export interface GameBrowseResultDto {
  gameId: string | null
  igdbId: number | null
  title: string
  coverImageUrl: string | null
  releaseDate: string | null
  platforms: string | null
  genres: string | null
}

export interface GameSummaryResponseDto {
  gameId: string
  igdbId: number | null
  title: string
  coverImageUrl: string | null
  releaseDate: string | null
  platforms: string | null
  genres: string | null
}

export interface GameDetailResponseDto {
  gameId: string
  igdbId: number | null
  title: string
  slug: string | null
  summary: string | null
  coverImageUrl: string | null
  releaseDate: string | null
  developer: string | null
  publisher: string | null
  platforms: string | null
  genres: string | null
  createdAt: string
  updatedAt: string
}

export interface ImportedGameResponseDto {
  gameId: string
  igdbId: number
  title: string
  slug: string | null
  summary: string | null
  coverImageUrl: string | null
  releaseDate: string | null
  developer: string | null
  publisher: string | null
  platforms: string | null
  genres: string | null
  updatedAt: string
}

export interface GetGamesParams {
  query?: string
  take?: number
}